services/ui-src/src/components/fields/ChoiceListField.test.tsx

```diff
@@ -252,7 +288,53 @@ describe("Test ChoiceListField hydration functionality", () => {
   });
 });
 
+describe("Test Choicelist component autosaves", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("Choicelist Checkbox autosaves with checked value when stateuser, autosave true, and form is valid", async () => {
+    mockedUseUser.mockReturnValue(mockStateUser);
+    mockGetValues(undefined);
+    const mockHydrationValue = [{ key: "Choice 1", value: "Choice 1" }];
+    render(CheckboxComponentAutosave);
+    const checkBox1 = screen.getByText("Choice 1");
+    expect(checkBox1).toBeVisible();
+
+    await act(async () => {
+      await userEvent.click(checkBox1);
+    });
+
+    expect(mockSetValue).toHaveBeenCalledWith(
+      "autosaveCheckboxField",
+      mockHydrationValue,
+      {
+        shouldValidate: true,
+      }
+    );
+
+    expect(mockReportContext.updateReport).toHaveBeenCalledTimes(1);
+    expect(mockReportContext.updateReport).toHaveBeenCalledWith(
+      {
+        state: mockStateUser.user?.state,
+        id: mockReportContext.report.id,
+      },
+      {
+        status: ReportStatus.IN_PROGRESS,
+        lastAlteredBy: mockStateUser.user?.full_name,
+        fieldData: {
+          autosaveCheckboxField: mockHydrationValue,
+        },
+      }
+    );
+  });
```

Comment:
I think we should also test that it doesn't autosave if its not a stateuser or if autosave isn't true

Reply:
Updated 👍